lib/puma/plugin/statsd.rb

```diff
@@ -86,25 +27,46 @@ def max_threads
   # We can start doing something when we have a launcher:
   def start(launcher)
     @launcher = launcher
-
-    @statsd = ::StatsdConnector.new
-    if @statsd.enabled?
-      @launcher.events.debug "statsd: enabled (host: #{@statsd.host})"
-
-      # Fetch global metric prefix from env variable
-      @metric_prefix = ENV.fetch("STATSD_METRIC_PREFIX", nil)
-      if @metric_prefix && !@metric_prefix.end_with?(::StatsdConnector::METRIC_DELIMETER)
-        @metric_prefix += ::StatsdConnector::METRIC_DELIMETER
-      end
+    host = ENV.fetch("STATSD_HOST", nil)
+    port = ENV.fetch("STATSD_PORT", 8125)
+    
+    if host
+      @statsd_client = Datadog::Statsd.new(host, port, tags: environment_variable_tags)
+      @launcher.events.debug "statsd: enabled (host: #{host})"
 
       register_hooks
     else
-      @launcher.events.debug "statsd: not enabled (no #{StatsdConnector::ENV_NAME} env var found)"
+      @launcher.events.debug "statsd: not enabled (no STATSD_HOST env var found)"
     end
   end
 
   private
 
+  # Send data to statsd every few seconds
+  def stats_loop
+    sleep 5
```

Comment:
should this be configurable?

Reply:
I think so. This is the initial delay, and [here](https://github.com/runtastic/puma-plugin-statsd/pull/2/files#diff-66ae24fc7c894553874428f3778a1eac3798df59f43a30a7c33e2f5007608a37R65) we have the reporting interval. But tbh I really don't know how to pass configurations for puma plugins, and I wanted to release a first version ASAP, so maybe we can invest in this in a later time.

Reply:
sure this can be a later update. and config could just be done via env variables (while having some defaults to fallback)